app/controllers/mixins/middleware_deployments_mixin.rb

```diff
@@ -16,9 +16,9 @@ def add_deployment
 
   def find_existing_deployment
     if @is_server
-      MiddlewareDeployment.find_by(:name => @deployment_name, :server_id => @entity_id)
+      MiddlewareDeployment.find_by(:name => @deployment_name, :server_id => from_cid(@entity_id))
     else
-      MiddlewareDeployment.find_by(:name => @deployment_name, :server_group_id => @entity_id)
+      MiddlewareDeployment.find_by(:name => @deployment_name, :server_group_id => from_cid(@entity_id))
```

Comment:
It's defined in [app/controllers/mixins/compressed_ids.rb](https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/controllers/mixins/compressed_ids.rb) which is included in application_helper. I think it's an easy way of fetching correct IDs, since we can get both full Number ID `10000000000055` and shorten ID `10r55`.